component/admin/updates/3.0.3.php

```diff
@@ -18,6 +18,44 @@
  */
 class RedshopUpdate303 extends RedshopInstallUpdate
 {
+	/**
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function installShopperGroup()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('*')
+			->from($db->qn('#__redshop_tax_rate'));
+		$taxRates = $db->setQuery($query)->loadAssocList();
+
+		foreach ($taxRates as $taxRate) {
+			$subQuery = $db->getQuery(true)
+				->select('*')
+				->from($db->qn('#__redshop_tax_shoppergroup_xref'))
+				->where($db->qn('tax_rate_id') . ' = ' . $db->q($taxRate['id']));
+			$taxShopperGroups = $db->setQuery($subQuery)->loadAssocList();
+
+			if (!$taxShopperGroups) {
+				$query = $db->getQuery(true)
+					->select('id')
+					->from($db->qn('#__redshop_shopper_group'));
+
+				$shopperGroups = $db->setQuery($query)->loadAssocList();
+
+				foreach ($shopperGroups as $shopperGroup) {
+					$obTaxShopperGroup = new stdClass();
+					$obTaxShopperGroup->tax_rate_id = $taxRate['id'];
+					$obTaxShopperGroup->shopper_group_id = $shopperGroup['id'];
+					$db->insertObject('#__redshop_tax_shoppergroup_xref', $obTaxShopperGroup, 'id');
+				}
+			}
+		}
+	}
+
```

Comment:
please optimize code and query, too much queries here that causing very bad performance